### PR TITLE
[25.1] Sequence datatypes: only read first character per line for setting metadata

### PR DESCRIPTION
**Pull request by bernt-matthias**
otherwise the whole line is kept in memory which
might be the complete sequence in edge cases

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/datatypes/sequence.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> N
         data_lines = 0
         sequences = 0
         with compression_utils.get_fileobj(dataset.get_file_name()) as fh:
-            for line in fh:
+            for line in iter_start_of_line(fh, 1):
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/sequence.py:380]:
```
F821 Undefined name `iter_start_of_line`
   --> lib/galaxy/datatypes/sequence.py:379:25
    |
377 |         sequences = 0
378 |         with compression_utils.get_fileobj(dataset.get_file_name()) as fh:
379 |             for line in iter_start_of_line(fh, 1):
```

**Comment by mvdbeek** [comment]
Can you directly target 25.1 ?

File: lib/galaxy/datatypes/sequence.py [hunk 2]
```diff
@@ -376,7 +377,7 @@ def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> N
...
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/sequence.py:380]: We do the same / similar things in many places, but `BaseFastq` and `Sequence` and `Tabular` seem like high value places that are worth fixing too

Comment (bernt-matthias) [reply lib/galaxy/datatypes/sequence.py:380]: I will go through the sequence data types and apply the same. `Tabular` is more tricky, since we count number of columns and get column types (but this can certainly by done also with less memory). I would not include it here.

Comment (mvdbeek) [reply lib/galaxy/datatypes/sequence.py:380]: `util.iter_start_of_line(fh, MAX_LINE_LEN)` ?

**Comment by bernt-matthias** [comment]
OK rebased to 25.1
